operations/helm/charts/mimir-distributed/templates/ruler-query-scheduler/ruler-query-scheduler-dep.yaml

```diff
@@ -0,0 +1,129 @@
+{{- if .Values.ruler_querier_service.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "mimir.resourceName" (dict "ctx" . "component" "ruler-query-scheduler") }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+    {{- include "mimir.labels" (dict "ctx" . "component" "ruler-query-scheduler") | nindent 4 }}
+  annotations:
+    {{- toYaml .Values.ruler_query_scheduler.annotations | nindent 4 }}
+spec:
+  replicas: {{ .Values.ruler_query_scheduler.replicas }}
+  selector:
+    matchLabels:
+      {{- include "mimir.selectorLabels" (dict "ctx" . "component" "ruler-query-scheduler") | nindent 6 }}
+  strategy:
+    {{- toYaml .Values.ruler_query_scheduler.strategy | nindent 4 }}
+  template:
+    metadata:
+      labels:
+        {{- include "mimir.podLabels" (dict "ctx" . "component" "ruler-query-scheduler") | nindent 8 }}
+      annotations:
+        {{- include "mimir.podAnnotations" (dict "ctx" . "component" "ruler-query-scheduler") | nindent 8 }}
+    spec:
+      serviceAccountName: {{ template "mimir.serviceAccountName" . }}
+      {{- if .Values.ruler_query_scheduler.priorityClassName }}
+      priorityClassName: {{ .Values.ruler_query_scheduler.priorityClassName }}
+      {{- end }}
+      securityContext:
+        {{- include "mimir.lib.podSecurityContext" (dict "ctx" . "component" "ruler-query-scheduler") | nindent 8 }}
+      initContainers:
+        {{- toYaml .Values.ruler_query_scheduler.initContainers | nindent 8 }}
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range .Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end}}
+      {{- end }}
+      containers:
+        - name: ruler-query-scheduler
+          image: "{{ include "mimir.imageReference" . }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            - "-target=query-scheduler"
+            - "-config.expand-env=true"
+            - "-config.file=/etc/mimir/mimir.yaml"
+            - "-server.grpc.keepalive.max-connection-age=2562047h" # 100000 days, effectively infinity
+            - "-server.grpc.keepalive.max-connection-age-grace=2562047h" # 100000 days, effectively infinity
```

Comment:
What is the purpose of setting such a high value? Seems unsafe since you could leak connections over time.

Reply:
@Logiraptor  these are brought over from our query-scheduler templates. What are reasonable values ?

Reply:
Gotcha, there could be a reason for this. Let me look into it.

Reply:
Found my answer here: https://github.com/grafana/mimir/pull/3262